README.md

```diff
@@ -1,3 +1,164 @@
+of how the output depends on the transformation parameters is accessed via
+`transform_deriv_params(trans, x)`. Users currently have to overload these methods,
+as no fall-back automatic differentiation is currently included. Alternatively,
+all the built-in types and transformations are compatible with automatic differentiation
+techniques, and can be parameterized by *DualNumbers*' `DualNumber` or *ForwardDiff*'s `GradientNumber`.
+
+### Built-in transformations
+
+A small number of 2D and 3D coordinate systems and transformations are included.
+We also have `IdentityTransform{InOutType}` and `ComposedTransformation{InType,OutType}`,
+which allows us to nest together arbitrary transformations to create a
+complex yet efficient transformation chain.
+
+#### Coordinate system types
+
+The canonical Cartesian coordinate systems are any type which inherited from
+*FixedSizeArrays*' `FixedVector{2}` and `FixedVector{3}` in 2D and 3D,
+respectively. The concrete datatypes `Point` and `Vec` are provided by
+*FixedSizeArrays*, and here we tend to favor `Point` for output where it is not
+specified. (Of course, users may define their own transformations and types, as
+they wish).
+
+The `Polar(r, θ)` type is a 2D polar representation of a point, and similarly
+in 3D we have defined `Spherical(r, θ, ϕ)` and `Cylindrical(r, θ, z)`.
+
+
+#### Coordinate system transformations
+
+Two-dimensional coordinates may be converted using these parameterless
+transformations:
+
+1. `PolarFromCartesian()`
+2. `CartesianFromPolar()`
+
+Three-dimensional coordinates may be converted using these singleton
+transformations:
+
+1. `SphericalFromCartesian()`
+2. `CartesianFromSpherical()`
+3. `SphericalFromCylindrical()`
+4. `CylindricalFromSpherical()`
+5. `CartesianFromCylindrical()`
+6. `CylindricalFromCartesian()`
+
+#### Translations
+
+Translations can be be applied to Cartesian coordinates in arbitrary dimensions,
+by e.g. `Translation(Δx, Δy)` or `Translation(Δx, Δy, Δz)` in 2D/3D.
+
+#### Rotations
+
+Rotations in 2D and 3D are treated slightly differently. The `Rotation2D(Δθ)` rotates a
+Cartesian point about the origin, while `RotationPolar(Δθ)` rotates a `Polar`
+coordinate about the origin.
+
+`Rotation` performs rotations of 3D Cartesian coordinates. Rotations may be
+defined simply by their transformation matrix (assumed to be orthogonal),
+or via any of the types exported from the *Rotations* package, including
+`Quaternion` (originating from the *Quaternions* package), `AngleAxis`,
+`EulerAngles`, `ProperEulerAngles` and `SpQuat`. In these latter cases, the
+"parameters" differentiated by `transform_deriv_params()` come directly from the
+parameterization of the quaternion, Euler angles, etc. **Note:** the derivative
+is only guaranteed to be correct in the tangent plane of the constrained variable,
+such as normalized quaternions, and optimization techniques should take this
+into account as necessary.
```

Comment:
Nice.  I tried to make this point clearer, but didn't come up with anything too compelling.